solidity/contracts/deposit/DepositFunding.sol

```diff
@@ -96,14 +97,21 @@ library DepositFunding {
     }
 
     /// @notice     Anyone may notify the contract that signing group setup has timed out.
-    /// @dev        We rely on the keep system to punish the signers in this case.
     /// @param  _d  Deposit storage pointer.
     function notifySignerSetupFailure(DepositUtils.Deposit storage _d) public {
         require(_d.inAwaitingSignerSetup(), "Not awaiting setup");
         require(
             block.timestamp > _d.signingGroupRequestedAt.add(TBTCConstants.getSigningGroupFormationTimeout()),
             "Signing group formation timeout not yet elapsed"
         );
+
+        // refund the deposit owner the cost to create a new keep at current price levels.
+        uint256 _seized = _d.seizeSignerBonds();
+
+        /* solium-disable-next-line security/no-send */
+        _d.depositOwner().send(_d.keepSetupFee);
```

Comment:
If possible, this should be a pull payment.

Reply:
I'm going to merge this branch and have us consider the pull payment aspect holistically across the codebase next week.

Reply:
This landed in #570.